lib/rack/mock_response.rb

```diff
@@ -11,6 +10,41 @@ module Rack
   # MockRequest.
 
   class MockResponse < Rack::Response
+    # Compatibility around CGI::Cookie for Ruby >= 3.5
+    class Cookie # :nodoc:
+      attr_reader :name, :value, :path, :domain, :expires, :secure
+
+      def initialize(args)
+        @name = args["name"]
+        @value = args["value"]
+        @path = args["path"]
+        @domain = args["domain"]
+        @expires = args["expires"]
+        @secure = args["secure"]
+      end
+
+      def method_missing(method_name, *args, &block)
+        @value.send(method_name, *args, &block)
+      end
+      # :nocov:
+      ruby2_keywords(:method_missing) if respond_to?(:ruby2_keywords, true)
+      # :nocov:
+
+      def respond_to_missing?(method_name, include_all = false)
+        @value.respond_to?(method_name, include_all) || super
+      end
+    end
+    private_constant :Cookie 
```

Comment:
I think it is a bad idea to use `private_constant` for this class when you could be returning instances of it.  I would only define the class on Ruby 3.5+, and have `Rack::MockResponse::Cookie` be this class on Ruby 3.5+, and `CGI::Cookie` on earlier Ruby versions.

Reply:
Makes it a bit nicer actually, thanks. I ran external tests on my fork and grape fails because it defines `Rack::MockResponse::Cookie` already (in test only!), so I openend https://github.com/ruby-grape/grape/pull/2565 for that.